bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/CreateNewLogTest.java

```diff
@@ -142,6 +151,237 @@ public void testCreateNewLogWithNoWritableLedgerDirs() throws Exception {
+
+        // Creating a new configuration with a number of ledger directories.
+        conf.setLedgerDirNames(ledgerDirs);
+        // forceGCAllowWhenNoSpace is disabled
+        conf.setIsForceGCAllowWhenNoSpace(false);
+        // pre-allocation is not enabled
+        conf.setEntryLogFilePreAllocationEnabled(false);
+        conf.setEntryLogPerLedgerEnabled(true);
+        LedgerDirsManager ledgerDirsManager = new LedgerDirsManager(conf, conf.getLedgerDirs(),
+                new DiskChecker(conf.getDiskUsageThreshold(), conf.getDiskUsageWarnThreshold()));
+        EntryLogger entryLogger = new EntryLogger(conf, ledgerDirsManager);
+        EntryLoggerAllocator entryLoggerAllocator = entryLogger.entryLoggerAllocator;
+        EntryLogManagerForEntryLogPerLedger entryLogManager = (EntryLogManagerForEntryLogPerLedger)
+                entryLogger.getEntryLogManager();
+
+        Thread.sleep(200);
+        int expectedPreAllocatedLogIDDuringInitialization = -1;
+        Assert.assertEquals("PreallocatedlogId after initialization of Entrylogger",
+                expectedPreAllocatedLogIDDuringInitialization, entryLoggerAllocator.getPreallocatedLogId());
+        Assert.assertEquals("Preallocation Future of this slot should be null", null,
+                entryLogger.entryLoggerAllocator.preallocation);
+
+        long ledgerId = 0L;
+
+        entryLogManager.createNewLog(ledgerId);
+        Thread.sleep(100);
+        /*
+         * pre-allocation is not enabled, so it would not preallocate for next entrylog
+         */
+        Assert.assertEquals("PreallocatedlogId after initialization of Entrylogger",
+                expectedPreAllocatedLogIDDuringInitialization + 1, entryLoggerAllocator.getPreallocatedLogId());
+
+        for (int i = 0; i < numDirs - 1; i++) {
+            ledgerDirsManager.addToFilledDirs(Bookie.getCurrentDirectory(new File(ledgerDirs[i])));
+        }
+
+        /*
+         * this is the only non-filled ledgerDir so it should be used for creating new entryLog
+         */
+        File nonFilledLedgerDir = Bookie.getCurrentDirectory(new File(ledgerDirs[numDirs - 1]));
+
+        entryLogManager.createNewLog(ledgerId);
+        BufferedLogChannel newLogChannel = entryLogManager.getCurrentLogForLedger(ledgerId);
+        Assert.assertEquals("Directory of newly created BufferedLogChannel file", nonFilledLedgerDir.getAbsolutePath(),
+                newLogChannel.getLogFile().getParentFile().getAbsolutePath());
+
+        ledgerDirsManager.addToFilledDirs(Bookie.getCurrentDirectory(new File(ledgerDirs[numDirs - 1])));
+
+        // new entrylog creation should succeed, though there is no writable ledgerDir
+        entryLogManager.createNewLog(ledgerId);
+    }
+
+    /*
+     * In this testcase it is validated if the entryLog is created in the
+     * ledgerDir with least number of current active entrylogs
+     */
+    @Test(timeout = 60000)
```

Comment:
IIRC we don't need "(timeout = 60000)" anymore

Reply:
yes, will remove